### PR TITLE
Make Horde.Supervisor child_spec overridable

### LINKED ISSUE #107: Improve module based supervisor
Current module based supervisor implementation does not allow setting supervisor options like max restarts, timers and so on, that's because of:

* child_spec is not overridable
* init/1 in standard module-based supervisor is expected to return a Supervisor.init call, in horde returns some options that's not clear what are used for

My suggestion is to make the module-based supervisor as the standard one. If is too complicated maybe make child_spec overridable in the macro?

### PR DESCRIPTION
**Pull request by ordepdev**
Fixes #107

### CODE REVIEW

**Comment by ordepdev** [comment]
Hmm, running the tests locally it passes with:
```
Finished in 37.9 seconds
3 properties, 70 tests, 0 failures, 1 skipped
```

Still, is failing in circleci with:

```
1) test .set_members/2 can join and unjoin registry with set_members (ClusterTest)
     test/cluster_test.exs:164
     match (=) failed
     code:  assert 1 = Enum.count(members)
     right: 2
     stacktrace:
       test/cluster_test.exs:190: (test)

Finished in 37.6 seconds
3 properties, 70 tests, 1 failure, 1 skipped
```

**Comment by ordepdev** [comment]
Oh, it was my mistake 😅

**Comment by derekkraan** [comment]
Also a good chance it was a wonky test! There are a couple of those here.

File: test/supervisor_test.exs [hunk 1]
```diff
@@ -89,6 +89,28 @@ defmodule SupervisorTest do
...
+           [
+             custom_id: 123,
+             name: :init_sup_test_3,
+             strategy: :one_for_one
+           ]},
+        restart: :transient,
+        type: :supervisor
+      }
+
+      assert spec =
```
Comment (derekkraan) [inline test/supervisor_test.exs]: This assertion isn't asserting anything I don't think. Should be `^spec` I think.

Comment (ordepdev) [reply test/supervisor_test.exs]: Ohhh, it was testing... nothing  🙈 Already fixed!

**Comment by derekkraan** [comment]
Looks great, thanks!!

**Pull request merged**